posydon/popsyn/rate_calculation.py

```diff
@@ -689,20 +703,230 @@ def compute_merger_rate_density(self, w_ijk, z_event, observable='DCOs', sensiti
-                return Rate_DCOs # Gpc^-3 yr^-1
+                return Rate_DCO # Gpc^-3 yr^-1
             else:
                 raise ValueError('Unsupported sensitivity!')
+        
+        elif observable in ['GRB1','GRB2']:
+            s = observable[-1]
+            z_GRB = z_event
+            if sensitivity=='beamed':
+                if index is not None:
+                    f_beaming = self.get_data(f"S{s}_f_beaming",index)
+                    flag_GRB = self.get_data(f"GRB{s}",index)
+                else:
+                    raise ValueError('Missing f_beaming parameter!')
+                Rate_GRB = np.zeros(n-1)
+                for i in range(1,n):
+                    cond_GRB = np.logical_and(np.logical_and(z_GRB>z_hor[i-1], z_GRB<=z_hor[i]),
+                                              flag_GRB)
+                    f_fb = f_beaming[cond_GRB]
+                    Rate_GRB[i-1] = (sum(w_ijk[cond_GRB]*f_fb)
+                                      /self.get_shell_comovig_volume(z_hor[i-1],
+                                                                     z_hor[i],
+                                                                     sensitivity='infinite'))
+                return Rate_GRB
+
+            elif sensitivity=='infinite':
+                if index is not None:
+                    flag_GRB = self.get_data(f"GRB{s}",index)
+                else:
+                    raise ValueError('Missing f_beaming parameter!')
+                Rate_GRB = np.zeros(n-1)
+                for i in range(1,n):
+                    if index is None:
+                        raise ValueError('Provide index comlumn to identify GRB systems.')
+                    cond_GRB = np.logical_and(np.logical_and(z_GRB>z_hor[i-1], z_GRB<=z_hor[i]),
+                                              flag_GRB)
+                    Rate_GRB[i-1] = (sum(w_ijk[cond_GRB])
+                                      /self.get_shell_comovig_volume(z_hor[i-1],
+                                                                     z_hor[i],
+                                                                     sensitivity))
+                return Rate_GRB
+            else:
+                raise ValueError('Unknown sensitivity!')
         else:
             raise ValueError('Unknown observable!')
+
+    ##############################
+    #####  GRB class methods #####
+    ##############################
+    
+    def get_time_GRB(self, z_birth, event=None):
+        """Get the time of the GRB.
+
+        Parameters
+        ----------
+        z_brith : double
+            Redshif of birth.
+        event : string
+            Event you are tracking, either first or second core collpase:
+            'CC1', 'CC2'.
+
+        Returns
+        -------
+        t_BRB : double
+            Cosmic time of the GRB event in Gyr.
+
+        """
+        if event not in ['CC1', 'CC2']:
+            raise ValueError(f'Unknown event {event}!')
```

Comment:
in the for loop on line 837 you specify `for s, event in enumerate(['CC1', 'CC2'])` why is this ValueError need here?

Reply:
To be safe, I am just preventing future misuse of this method which could be renamed to a more genric name.